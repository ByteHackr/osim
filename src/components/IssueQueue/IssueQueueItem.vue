<script setup lang="ts">
import { computed } from 'vue';

import { useToggle } from '@vueuse/core';

import type { FilteredIssue } from './IssueQueue.vue';

const { issue } = defineProps<{
  issue: FilteredIssue;
}>();

// Temporarily hiding 'Source' column to avoid displaying incorrect information.
// TODO: unhide it once final issue sources are defined. [OSIDB-2424]
//       and update the CSS for the column width in IssueQueue

const nonIdFields: Exclude<keyof FilteredIssue, 'id'>[] =
  ['impact', 'formattedDate', 'title', 'workflowState', 'owner'];

const hasBadges = computed(() => issue.embargoed || !!issue.labels?.length);
const truncatedLabelCount = 4;
const [showLabels, toggleShowLabels] = useToggle();
const sortedLabels = computed(() => issue.labels?.toSorted((a, b) => {
  if (
    (a.state === 'REQ' && b.state !== 'REQ')
    || (a.contributor && !b.contributor)
  ) {
    return -1;
  }
  if (
    (a.state !== 'REQ' && b.state === 'REQ')
    || (!a.contributor && b.contributor)
  ) {
    return 1;
  }

  return a.label.localeCompare(b.label);
}) ?? []);
const truncatedLabels = computed(
  () => showLabels.value ? sortedLabels.value : sortedLabels.value?.slice(0, truncatedLabelCount),
);
</script>

<template>
  <tr
    class="osim-issue-queue-item"
    :class="$attrs.class"
  >
    <td
      class="osim-issue-title"
      :class="{ 'pb-0': hasBadges }"
    >
      <RouterLink :to="{ name: 'flaw-details', params: { id: issue.id } }">
        {{ issue.id }}
      </RouterLink>
    </td>
    <td
      v-for="field in nonIdFields"
      :key="field"
      :class="{ 'pb-0': hasBadges }"
    >
      {{ issue[field] }}
    </td>
    <!--<td>{{ issue.assigned }}</td>-->
  </tr>
  <tr
    v-if="hasBadges"
    class="osim-badge-lane"
    :class="$attrs.class"
  >
    <td colspan="100%">
      <div class="gap-1 d-flex">
        <span
          v-if="issue.embargoed"
          class="badge rounded-pill text-bg-danger border border-primary"
        >Embargoed</span>
        <span
          v-for="label in truncatedLabels"
          :key="label.label"
          class="badge rounded-pill text-capitalize border"
          :class="{
            'text-bg-warning fw-bold border-warning': label.state == 'REQ',
            'text-bg-light-info border-info': label.state != 'REQ' && label.relevant,
            'text-decoration-line-through text-bg-gray border-secondary': !label.relevant,
          }"
          :title="label.state == 'REQ' ? 'Requested' : ''"
        >{{ label.label }}</span>
        <i
          v-if="(issue.labels?.length ?? 0) > truncatedLabelCount"
          class="bi pe-1 cursor-pointer osim-show-all-labels"
          :class="[showLabels ? 'bi-caret-left-fill' : 'bi-caret-right-fill']"
          title="Show all labels"
          @click="toggleShowLabels()"
        />
      </div>
    </td>
  </tr>
</template>

<style lang="scss" scoped>
@import '@/scss/bootstrap-overrides';

td.osim-issue-title {
  max-width: 10rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

tr td {
  border: 0;
}

tr.osim-shaded td {
  background-color: $light-gray;
}

tr:hover {
  td {
    background-color: #ffe3d9;
  }

  &.osim-issue-queue-item + tr.osim-badge-lane td {
    background-color: #ffe3d9;
  }
}

tr.osim-issue-queue-item:has(+ tr.osim-badge-lane:hover) td {
  background-color: #ffe3d9;
}

.osim-badge-lane {
  .osim-show-all-labels {
    cursor: pointer;
  }
}
</style>
