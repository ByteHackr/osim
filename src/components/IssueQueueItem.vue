<script setup lang="ts">
import { DateTime } from 'luxon';

const props = defineProps<{
  issue: any,
  selected: boolean,
}>();

defineEmits<{
  (e: 'update:selected', selected: boolean): void;
}>();
// console.log('IssueQueueItem:', props.issue);

</script>

<template>
  <tr class="osim-issue-queue-item">
    <td><input :checked="selected" @change="$emit('update:selected', ($event.target as HTMLInputElement).checked)" type="checkbox" class="form-check-input" aria-label="Select Issue"></td>
    <td>
      <RouterLink :to="{name: 'flaw-details', params: {id: issue.uuid}}">{{ issue.cve_id || issue.uuid }}</RouterLink>
      <div v-if="issue.unembargo_dt">
        <span v-if="DateTime.fromISO(issue.unembargo_dt).diffNow().milliseconds > 0">
          <span class="badge rounded-pill text-bg-danger">Embargoed</span>
        </span>
      </div>
    </td>
    <td>{{ issue.impact }}</td>
    <td>{{ issue.source }}</td>
    <td>{{ issue.created_dt }}</td>
    <td>{{ issue.title }}</td>
    <td>{{ issue.state }}</td>
    <td>{{ issue.owner }}</td>
    <!--<td>{{ issue.assigned }}</td>-->
  </tr>
</template>

<style scoped>

</style>
