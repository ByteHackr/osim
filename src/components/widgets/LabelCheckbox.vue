<script setup lang="ts">
const props = defineProps<{
  modelValue: boolean,
  label: string,
}>();

</script>

<template>
  <div class="osim-input border-start ps-3">
  <label class="form-check mb-3">
    <input
        type="checkbox"
        :checked="modelValue"
        class="form-check-input"
        v-bind="$attrs"
        :value="modelValue"
        @change="$emit('update:modelValue', (($event as InputEvent).target as HTMLInputElement).checked)"
    />
    <span class="form-check-label">
      {{ label }}
    </span>
  </label>
  </div>
</template>

<style scoped>

.osim-input {
  display: block;
}

</style>
