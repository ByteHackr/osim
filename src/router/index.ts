import {createRouter, createWebHistory} from 'vue-router'
import {nextTick as vueNextTick} from 'vue';
import IndexView from '../views/IndexView.vue'
import LoginView from '../views/LoginView.vue';
import FlawDetailView from '../views/FlawDetailView.vue';
import {useUserStore} from '@/stores/UserStore';
import NotFoundView from '@/views/NotFoundView.vue';
import TrackerDetailView from '@/views/TrackerDetailView.vue';

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: '/',
      name: 'index',
      component: IndexView,
      meta: {
        title: 'Index',
      },
    },
    {
      path: '/flaws/:id',
      name: 'flaw-details',
      component: FlawDetailView,
      props: true,
      meta: {
        title: 'Flaw Details',
      },
    },
    {
      path: '/login',
      name: 'login',
      component: LoginView,
      meta: {
        title: 'Login',
        hideNavbar: true,
      },
    },

    {
      path: '/tracker/:id',
      name: 'tracker-details',
      props: true,
      component: TrackerDetailView,
      meta: {
        title: 'Tracker Details',
      },
    },

    // {
    //   path: '/flaw-details',
    //   name: 'flaw-details',
    //   component: FlawView,
    // },

    {
      path: '/:pathMatch(.*)*',
      name: 'not-found',
      component: NotFoundView,
      meta: {
        title: 'Page Not Found',
      },
    },

    // {
    //   path: '/about/:id',
    //   name: 'about',
    //   // route level code-splitting
    //   // this generates a separate chunk (About.[hash].js) for this route
    //   // which is lazy-loaded when the route is visited.
    //   component: () => import('../views/AboutView.vue')
    //   meta: {
    //     title: 'About',
    //   },
    // },
  ]
})


router.beforeEach((to, from) => {
  const {isAuthenticated} = useUserStore();

  if (isAuthenticated()) {
    if (to.name === 'login') {
      return false;
    }
  } else {
    if (to.name !== 'login') {
      return {name: 'login'};
    }
  }

router.afterEach((to, from) => {
  vueNextTick(() => {
    document.title = 'OSIM | ' + to.meta.title;
  })
})

export default router
