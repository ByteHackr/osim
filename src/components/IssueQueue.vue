<script setup lang="ts">
import { computed, onMounted, reactive, ref, onUnmounted, watch } from 'vue';
import { DateTime } from 'luxon';
import IssueQueueItem from '@/components/IssueQueueItem.vue';
import LabelCheckbox from './widgets/LabelCheckbox.vue';
import { useUserStore } from '@/stores/UserStore';
import { FlawClassificationStateEnum } from '../generated-client';
const userStore = useUserStore();

const emit = defineEmits(['flaws:fetch', 'flaws:load-more']);

type FilteredIssue = {
  issue: any;
  selected: boolean;
};

type ColumnField = 'id' | 'impact' | 'source' | 'created_dt' | 'title' | 'state' | 'owner';

const props = defineProps<{
  issues: any[];
  isLoading: boolean;
  isFinalPageFetched: boolean;
}>();

const issues = computed<any[]>(() => props.issues.map(relevantFields));
const selectedSortField = ref<ColumnField>('created_dt');
const isSortedByAscending = ref(false);
const isMyIssuesSelected = ref(false);
const isOpenIssuesSelected = ref(false);
const tableContainerEl = ref<HTMLElement | null>(null);

const filteredStates = computed(() => {
  const allStates = Object.values(FlawClassificationStateEnum);
  return allStates
    .filter(
      (state) =>
        state !== FlawClassificationStateEnum.Done &&
        state !== FlawClassificationStateEnum.Rejected,
    )
    .join(',');
});

const params = computed(() => {
  const paramsObj: Record<string, any> = {};

  if (isMyIssuesSelected.value) {
    paramsObj.owner = userStore.userName;
  }

  if (isOpenIssuesSelected.value) {
    paramsObj.workflow_state = filteredStates.value;
  }

  const sortOrderPrefix = isSortedByAscending.value ? '' : '-';
  paramsObj.order = {
    [selectedSortField.value]: `${sortOrderPrefix}${selectedSortField.value}`,
    id: `${sortOrderPrefix}cve_id,${sortOrderPrefix}uuid`,
    state: `${sortOrderPrefix}workflow_state`,
  }[selectedSortField.value];
  return paramsObj;
});

const columnsFieldsMap: Record<string, ColumnField> = {
  ID: 'id',
  Impact: 'impact',
  Source: 'source',
  Created: 'created_dt',
  Title: 'title',
  State: 'state',
  Owner: 'owner',
};

const relevantIssues = computed<FilteredIssue[]>(() => {
  return issues.value
    .map((issue) => reactive({ issue, selected: false }));
});

function updateSelectAll(selectedAll: boolean) {
  for (const filteredIssue of relevantIssues.value) {
    filteredIssue.selected = selectedAll;
  }
}

function selectSortField(field: ColumnField) {
  isSortedByAscending.value =
    selectedSortField.value === field ? !isSortedByAscending.value : false;
  selectedSortField.value = field;
}

const isSelectAllIndeterminate = computed(() => {
  if (relevantIssues.value.length === 0) {
    return false;
  }
  return relevantIssues.value.some((it) => it.selected !== relevantIssues.value[0].selected);
});

const isSelectAllChecked = computed(() => {
  return relevantIssues.value.every((it) => it.selected);
});

function relevantFields(issue: any) {
  return {
    id: issue.cve_id || issue.uuid,
    impact: issue.impact,
    source: issue.source,
    created_dt: issue.created_dt,
    title: issue.title,
    workflowState: issue.classification.state,
    unembargo_dt: issue.unembargo_dt,
    embargoed: issue.embargoed,
    owner: issue.owner,
    formattedDate: DateTime.fromISO(issue.created_dt).toFormat('yyyy-MM-dd'),
  };
}

onMounted(() => {
  tableContainerEl.value?.addEventListener('scroll', handleScroll); // AutoScroll Option
  emit('flaws:fetch', params);
});

onUnmounted(() => {
  tableContainerEl.value?.removeEventListener('scroll', handleScroll);
});

// AutoScroll Method. Maybe have this as a user configurable option in the future?
function handleScroll() {
  if (props.isLoading || !tableContainerEl.value) return; // Do not load more if already loading

  const totalHeight = tableContainerEl.value.scrollHeight;
  const scrollPosition = tableContainerEl.value.scrollTop + tableContainerEl.value.clientHeight;
  // Trigger loading more content when the user has scrolled to 99% of the container's height, 
  if (scrollPosition >= totalHeight * 0.99) { 
    emitLoadMore();
  }
}

function emitLoadMore() {
  emit('flaws:load-more', params);
}

watch(params, () => {
  emit('flaws:fetch', params);
});
</script>

<template>
  <div class="osim-content container osim-issue-queue">
    <div class="osim-incident-filter">
      <LabelCheckbox v-model="isMyIssuesSelected" label="My Issues" class="d-inline-block" />
      <LabelCheckbox v-model="isOpenIssuesSelected" label="Open Issues" class="d-inline-block" />

      <span
        v-if="isLoading"
        class="spinner-border spinner-border-sm d-inline-block ms-3"
        role="status"
      >
        <span class="visually-hidden">Loading...</span>
      </span>
      <span v-if="isLoading"> Loading Flaws&hellip; </span>
    </div>
    <div ref="tableContainerEl" class="osim-incident-list">
      <table class="table align-middle" :class="{ 'osim-table-loading': isLoading }">
        <thead class="sticky-top">
          <!-- <thead class=""> -->
          <tr>
            <th>
              <input
                type="checkbox"
                class="form-check-input"
                :indeterminate="isSelectAllIndeterminate"
                :checked="isSelectAllChecked"
                aria-label="Select All Issues in Table"
                @change="updateSelectAll(($event.target as HTMLInputElement).checked)"
              />
            </th>
            <th
              v-for="(field, columnName) in columnsFieldsMap"
              :key="columnName"
              :class="{ 'osim-issue-queue-table-created': columnName === 'Created' }"
              @click="selectSortField(field)"
            >
              {{ columnName }}
              <i
                :class="{
                  'opacity-0': selectedSortField !== field,
                  'bi-caret-up-fill': isSortedByAscending,
                  'bi-caret-down-fill': !isSortedByAscending,
                }"
                class="bi"
              >
              </i>
            </th>
          </tr>
        </thead>
        <tbody class="table-group-divider">
          <template v-for="(relevantIssue, index) of relevantIssues" :key="relevantIssue.id">
            <IssueQueueItem
              v-model:selected="relevantIssue.selected"
              :issue="relevantIssue.issue"
              :class="{
                'osim-shaded': index % 2 === 0,
              }"
            />
          </template>
        </tbody>
      </table>
      <button
        v-if="relevantIssues.length !== 0 && !isFinalPageFetched"
        class="btn btn-primary"
        type="button"
        :disabled="isLoading"
        @click="emitLoadMore"
      >
        <span
          v-if="isLoading"
          class="spinner-border spinner-border-sm d-inline-block"
          role="status"
        >
          <span class="visually-hidden">Loading...</span>
        </span>
        <span v-if="isLoading"> Loading Flaws&hellip; </span>
        <span v-else> Load More Flaws </span>
      </button>
      <p v-if="relevantIssues.length === 0">No results.</p>
      <span v-if="isFinalPageFetched" role="status">Nothing else to load.</span>
    </div>
  </div>
</template>

<style scoped lang="scss">
@import '@/scss/redhat-brand-colors';

.osim-issue-queue {
  font-family: 'Red Hat Mono', monospace;

  // tbody

  div.osim-incident-list {
    display: block;
    max-height: calc(100vh - 164px);
    overflow-y: scroll;

    &:hover::-webkit-scrollbar-thumb {
      background-color: $redhat-red-50;
    }
  }

  .osim-incident-filter {
    margin-left: -0.5rem;
  }

  &:hover i {
    color: $redhat-red-50;
  }

  table {
    &.osim-table-loading {
      opacity: 0.5;
    }

    th {
      cursor: pointer;
      user-select: none;
    }

    tr td,
    tr th {
      &:not(:first-child) {
        min-width: 12ch;
      }
    }

    tbody:hover::-webkit-scrollbar-thumb {
      background-color: $redhat-red-50;
    }
  }
}
</style>
