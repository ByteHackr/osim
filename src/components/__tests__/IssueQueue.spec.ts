import { mount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import router from '@/router';
import { createTestingPinia } from '@pinia/testing';
import IssueQueue from '../IssueQueue.vue';
import IssueQueueItem from '@/components/IssueQueueItem.vue';
import LabelCheckbox from '../widgets/LabelCheckbox.vue';

vi.mock('@vueuse/core', () => ({
  useLocalStorage: vi.fn((key: string) => {
    return {
      UserStore: {
        value: {
          // Set your fake user data here
          refresh: 'mocked_refresh_token',
          env: 'mocked_env',
          whoami: {
            email: 'test@example.com',
            username: 'testuser',
          },
        },
      },
    }[key];
  }),
  useStorage: vi.fn((key: string, defaults) => {
    return {
      'OSIM::API-KEYS': {
        value: defaults || {
          bugzillaApiKey: '',
          jiraApiKey: '',
          showNotifications: false,
        },
      },
    }[key];
  }),
}));

vi.mock('jwt-decode', () => ({
  default: vi.fn(() => ({
    sub: '1234567890',
    name: 'Test User',
    exp: Math.floor(Date.now() / 1000) + 60 * 60 * 24 * 365,
  })),
}));

describe('IssueQueue', () => {
  const mockData = [
    {
      uuid: '709e9ea1-ed0f-49b8-a7ad-9164d4034849',
      cve_id: 'CVE-2903-0092',
      state: 'NEW',
      impact: 'MODERATE',
      title: 'title',
      unembargo_dt: '2024-02-01T00:00:00Z',
      source: 'XEN',
      embargoed: false,
      created_dt: '2021-07-29T14:50:50Z',
      updated_dt: '2024-03-15T08:53:06Z',
      classification: {
        workflow: 'DEFAULT',
        state: 'NEW',
      },
      owner: 'test@redhat.com',
    },
  ];

  afterAll(() => {
    vi.clearAllMocks();
  });

  it('should fetch data from API', async () => {
    const pinia = createTestingPinia({
      createSpy: vitest.fn,
      stubActions: false,
    });
    const wrapper = mount(IssueQueue, {
      props: {
        issues: mockData,
        isLoading: false,
        isFinalPageFetched: false,
      },
      global: {
        plugins: [pinia, router],
      },
    });

    const fetchEvents = wrapper.emitted('flaws:fetch');
    expect(fetchEvents[0][0]._value).toEqual({
      order: '-created_dt'
    });
    const issues = wrapper.findAllComponents(IssueQueueItem);
    expect(issues.length).toBe(1);
  });

  it('fetch data from API with specified parameters on MyFlaws', async () => {
    const pinia = createTestingPinia({
      createSpy: vitest.fn,
      stubActions: false,
    });
    const wrapper = mount(IssueQueue, {
      props: {
        issues: mockData,
        isLoading: false,
        isFinalPageFetched: false,
      },
      global: {
        plugins: [pinia, router],
      },
    });

    const myIssuesCheckbox = wrapper.findAllComponents(LabelCheckbox)[0];
    const myIssuesCheckboxEl = myIssuesCheckbox.find('input[type="checkbox"]');
    await myIssuesCheckboxEl.setValue(true);
    await wrapper.vm.$nextTick();

    const fetchEvents = wrapper.emitted('flaws:fetch');
    expect(fetchEvents[1][0]._value).toEqual({
      order: '-created_dt',
      owner: 'test@example.com',
    });

    const issues = wrapper.findAllComponents(IssueQueueItem);
    expect(issues.length).toBe(1);
  });

  it('fetch data from API with specified parameters on OpenFlaws', async () => {
    const pinia = createTestingPinia({
      createSpy: vitest.fn,
      stubActions: false,
    });
    const wrapper = mount(IssueQueue, {
      props: {
        issues: [],
        isLoading: false,
        isFinalPageFetched: false,
      },
      global: {
        plugins: [pinia, router],
      },
    });

    const openIssuesCheckox = wrapper.findAllComponents(LabelCheckbox)[1];
    const openIssuesCheckboxEl = openIssuesCheckox.find('input[type="checkbox"]');
    await openIssuesCheckboxEl.setValue(true);
    await wrapper.vm.$nextTick();

    const fetchEvents = wrapper.emitted('flaws:fetch');
    expect(fetchEvents[1][0]._value).toEqual({ 
      order: '-created_dt',
      workflow_state: 'NEW,TRIAGE,PRE_SECONDARY_ASSESSMENT,SECONDARY_ASSESSMENT',
     
    });
    const issues = wrapper.findAllComponents(IssueQueueItem);
    expect(issues.length).toBe(0);
  });

  it('fetch data from API with specified parameters on sort', async () => {
    const pinia = createTestingPinia({
      createSpy: vitest.fn,
      stubActions: false,
    });
    const wrapper = mount(IssueQueue, {
      props: {
        issues: [],
        isLoading: false,
        isFinalPageFetched: false,
      },
      global: {
        plugins: [pinia, router],
      },
    });
    const idColumn = wrapper.findAll('th').at(1);
    await idColumn.trigger('click');
    const fetchEvents = wrapper.emitted('flaws:fetch');
    expect(fetchEvents[1][0]._value).toEqual({ 
      order: '-cve_id,-uuid',
    });
    // change sort order
    await idColumn.trigger('click');
    expect(fetchEvents[1][0]._value).toEqual({ 
      order: 'cve_id,uuid',
    });
    const impactColumn = wrapper.findAll('th').at(2);
    await impactColumn.trigger('click');
    expect(fetchEvents[1][0]._value).toEqual({ 
      order: '-impact',
    });
  });

  it('shouldn\'t render useDefault filter button', async () => {
    const pinia = createTestingPinia({
      createSpy: vitest.fn,
      stubActions: false,
    });
    const wrapper = mount(IssueQueue, {
      props: {
        issues: mockData,
        isLoading: false,
        isFinalPageFetched: false,
        showFilter: false
      },
      global: {
        plugins: [pinia, router],
      },
    });
    const defaultFilterCheckbox = wrapper.findAllComponents(LabelCheckbox)[2];
    expect(defaultFilterCheckbox).toBeFalsy();
  });

  it('should render useDefault filter button', async () => {
    const pinia = createTestingPinia({
      createSpy: vitest.fn,
      stubActions: false,
    });
    const wrapper = mount(IssueQueue, {
      props: {
        issues: mockData,
        isLoading: false,
        isFinalPageFetched: false,
        showFilter: true
      },
      global: {
        plugins: [pinia, router],
      },
    });
    const defaultFilterCheckbox = wrapper.findAllComponents(LabelCheckbox)[2];
    expect(defaultFilterCheckbox.exists()).toBeTruthy();
  });
});
