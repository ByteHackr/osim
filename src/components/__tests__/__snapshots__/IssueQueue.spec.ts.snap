// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`issueQueue > should render flaw labels 1`] = `
"<div data-v-48717bcb="" class="osim-content container-fluid osim-issue-queue">
  <div data-v-48717bcb="" class="osim-incident-filter">
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">My Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Open Issues</span></label></div>
    <!--v-if--><span data-v-48717bcb="" class="float-end"> Loaded 1 of 10</span>
  </div>
  <div data-v-48717bcb="" class="osim-incident-list">
    <table data-v-48717bcb="" class="table align-middle">
      <thead data-v-48717bcb="" class="sticky-top">
        <!-- <thead class=""> -->
        <tr data-v-48717bcb="">
          <th data-v-48717bcb="">ID <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Impact <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Created <i data-v-48717bcb="" class="bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Title <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">State <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Owner <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
        </tr>
      </thead>
      <tbody data-v-48717bcb="" class="table-group-divider">
        <tr data-v-15d9c71d="" class="osim-issue-queue-item osim-shaded">
          <td data-v-15d9c71d="" class="osim-issue-title pb-0"><a data-v-15d9c71d="" href="/flaws/CVE-2903-0092" class="">CVE-2903-0092</a></td>
          <td data-v-15d9c71d="" class="pb-0">MODERATE</td>
          <td data-v-15d9c71d="" class="pb-0">2021-07-29</td>
          <td data-v-15d9c71d="" class="pb-0">title</td>
          <td data-v-15d9c71d="" class="pb-0">NEW</td>
          <td data-v-15d9c71d="" class="pb-0">test@redhat.com</td>
          <!--<td>{{ issue.assigned }}</td>-->
        </tr>
        <tr data-v-15d9c71d="" class="osim-badge-lane osim-shaded">
          <td data-v-15d9c71d="" colspan="100%">
            <div data-v-15d9c71d="" class="gap-1 d-flex">
              <!--v-if--><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-warning fw-bold text-decoration-line-through text-bg-gray" title="Requested">test-3</span><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-light-info text-decoration-line-through text-bg-gray" title="">test</span><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-light-info text-decoration-line-through text-bg-gray" title="">test-2</span>
              <!--v-if-->
            </div>
          </td>
        </tr>
      </tbody>
    </table><button data-v-48717bcb="" class="btn btn-primary" type="button">
      <!--v-if--><span data-v-48717bcb=""> Load More Flaws </span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`issueQueue > should truncate flaw labels 1`] = `
"<div data-v-48717bcb="" class="osim-content container-fluid osim-issue-queue">
  <div data-v-48717bcb="" class="osim-incident-filter">
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">My Issues</span></label></div>
    <div data-v-b2ceeac4="" data-v-48717bcb="" class="osim-input ps-3 d-inline-block d-inline-block"><label data-v-b2ceeac4="" class="form-check h-100 d-inline-block ps-4 p-1"><input data-v-b2ceeac4="" class="d-inline-block form-check-input" type="checkbox"><span data-v-b2ceeac4="" class="form-check-label">Open Issues</span></label></div>
    <!--v-if--><span data-v-48717bcb="" class="float-end"> Loaded 1 of 10</span>
  </div>
  <div data-v-48717bcb="" class="osim-incident-list">
    <table data-v-48717bcb="" class="table align-middle">
      <thead data-v-48717bcb="" class="sticky-top">
        <!-- <thead class=""> -->
        <tr data-v-48717bcb="">
          <th data-v-48717bcb="">ID <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Impact <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Created <i data-v-48717bcb="" class="bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Title <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">State <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
          <th data-v-48717bcb="">Owner <i data-v-48717bcb="" class="opacity-0 bi-caret-down-fill bi"></i></th>
        </tr>
      </thead>
      <tbody data-v-48717bcb="" class="table-group-divider">
        <tr data-v-15d9c71d="" class="osim-issue-queue-item osim-shaded">
          <td data-v-15d9c71d="" class="osim-issue-title pb-0"><a data-v-15d9c71d="" href="/flaws/CVE-2903-0092" class="">CVE-2903-0092</a></td>
          <td data-v-15d9c71d="" class="pb-0">MODERATE</td>
          <td data-v-15d9c71d="" class="pb-0">2021-07-29</td>
          <td data-v-15d9c71d="" class="pb-0">title</td>
          <td data-v-15d9c71d="" class="pb-0">NEW</td>
          <td data-v-15d9c71d="" class="pb-0">test@redhat.com</td>
          <!--<td>{{ issue.assigned }}</td>-->
        </tr>
        <tr data-v-15d9c71d="" class="osim-badge-lane osim-shaded">
          <td data-v-15d9c71d="" colspan="100%">
            <div data-v-15d9c71d="" class="gap-1 d-flex">
              <!--v-if--><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-light-info text-decoration-line-through text-bg-gray" title="">test-0</span><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-light-info text-decoration-line-through text-bg-gray" title="">test-1</span><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-light-info text-decoration-line-through text-bg-gray" title="">test-2</span><span data-v-15d9c71d="" class="badge rounded-pill text-capitalize text-bg-light-info text-decoration-line-through text-bg-gray" title="">test-3</span><i data-v-15d9c71d="" class="bi pe-1 cursor-pointer osim-show-all-labels bi-caret-right-fill" title="Show all labels"></i>
            </div>
          </td>
        </tr>
      </tbody>
    </table><button data-v-48717bcb="" class="btn btn-primary" type="button">
      <!--v-if--><span data-v-48717bcb=""> Load More Flaws </span>
    </button>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
